### DISCUSSION TITLE
custom SASL mechanism

### DISCUSSION THREAD

**Discussion by viper645** [xmppjs/xmpp.js#917] [Q&A] [answered]
How can I use custom SASL mechanism not user pass, Because I need to authenticate with 2 jwt tokens.

![Screenshot_2_censored_censored](https://user-images.githubusercontent.com/36043176/129493765-506bd157-c7ab-45f1-a59e-ff770b284fb3.jpg)

**Comment by sonnyp** [comment, marked as answer]
You need to write a sasl meechanism compatible with https://github.com/jaredhanson/js-sasl

Then you can do something like `client.sasl.use(mySASLMechansim)`
